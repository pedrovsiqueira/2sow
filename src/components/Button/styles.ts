import styled from 'styled-components';

export const StyledButton = styled.button`
  background: #ecf0f3;
  min-width: 164px;
  min-height: 64px;
  box-shadow: 18px 18px 30px #d1d9e6, -18px -18px 30px #ffffff;
  border-radius: 17px;
  font-size: 2.6rem;
  line-height: 110%;
  padding: 15px 6px;
  letter-spacing: 0.02em;
  font-variant: small-caps;

  :hover {
    background: #e2e6e9;
    box-shadow: 18px 18px 30px #d1d9e6, -18px -18px 30px #ffffff;
    border-radius: 17px;
  }

  :focus {
    background: #2a2a2a;
    box-shadow: 18px 18px 30px #d1d9e6, -18px -18px 30px #ffffff;
    border-radius: 17px;
    color: #ecf0f3;
  }
`;
