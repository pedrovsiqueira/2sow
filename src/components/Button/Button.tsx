import React, { ButtonHTMLAttributes } from 'react';
import { StyledButton } from './styles';

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement>;

const Button: React.FC<ButtonProps> = ({ children, ...rest }) => (
  <StyledButton data-testid="button" {...rest}>
    {children}
  </StyledButton>
);
export default Button;
