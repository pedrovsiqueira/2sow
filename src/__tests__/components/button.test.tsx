import React from 'react';
import Button from './../../components/Button/Button';
import { render, cleanup } from '@testing-library/react';

beforeEach(cleanup)

it('Renders a button checking props without crashing', () => {
  const { getByText } = render(<Button>My Button Test</Button>);
  expect(getByText('My Button Test')).toBeDefined()
});

it('Verifies if a button was changed', () => {
  const {asFragment} = render(<Button>My Button Test</Button>)
  expect(asFragment()).toMatchSnapshot()
})