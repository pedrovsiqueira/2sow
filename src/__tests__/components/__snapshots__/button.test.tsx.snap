// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verifies if a button was changed 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm odmaE"
    data-testid="button"
  >
    My Button Test
  </button>
</DocumentFragment>
`;
